components/_impl/workers/subprocess_rpc.py

```diff
@@ -102,17 +104,19 @@ class _TimeoutPIPE:
     _loop_lock = threading.Lock()
     _active_reads: typing.Dict[int, typing.Tuple[float, float]]
     _loop_cadence = 1  # second
+    _proc_id: typing.Optional[int] = None
 
     @classmethod
-    def singleton(cls) -> "_TimeoutPIPE":
+    def singleton(cls, proc_id) -> "_TimeoutPIPE":
```

Comment:
We shouldn't have multiple singletons. (And `cls._singleton = cls(proc_id)` will trample with multiple active workers, which is currently allowed.)